neo/Network/P2P/Payloads/Cosigner.cs

```diff
@@ -0,0 +1,73 @@
+﻿using Neo.Cryptography.ECC;
+using Neo.IO;
+using Neo.IO.Json;
+using System;
+using System.IO;
+using System.Linq;
+
+namespace Neo.Network.P2P.Payloads
+{
+    public class Cosigner : ISerializable
+    {
+        public UInt160 Account;
+        public WitnessScope Scopes;
+        public UInt160[] AllowedContracts;
+        public ECPoint[] AllowedGroups;
+
+        // This limits maximum number of AllowedContracts or AllowedGroups here
+        private int MaxSubitems = 16;
+
+        public int Size =>
+            /*Account*/             UInt160.Length +
+            /*Scopes*/              sizeof(WitnessScope) +
+            /*AllowedContracts*/    (Scopes.HasFlag(WitnessScope.CustomContracts) ? AllowedContracts.GetVarSize() : 0) +
+            /*AllowedGroups*/       (Scopes.HasFlag(WitnessScope.CustomGroups) ? AllowedGroups.GetVarSize() : 0);
+
+        void ISerializable.Deserialize(BinaryReader reader)
+        {
+            Account = reader.ReadSerializable<UInt160>();
+            Scopes = (WitnessScope)reader.ReadByte();
+            if (Scopes.HasFlag(WitnessScope.Global) && (Scopes & ~WitnessScope.Global) != 0)
```

Comment:
Maybe Global could be 0x00

Reply:
No. `WitnessScope` is a flag.

Reply:
Do you mean, if all the flags are not set, then it is considered `Global`?

Reply:
Yes, maybe is easier

Reply:
Done.

Reply:
In fact, this was going to be my next suggestion.. but it may be misleading for OR logic @erikzhang . Think about it, someone sets flag Global and Entry, so Global should prevail, correct? If it's 0x80, Global will prevail, but if it's 0x00, Entry will prevail (like an AND behavior).
At the same time, if we combine Entry with Custom, both would work, as an OR, so the only AND possible would be with Global, not with others, which is strange... I think we may need to revert to 0x80.